lib/internal/test_runner/runner.js

```diff
@@ -560,7 +561,7 @@ function run(options = kEmptyObject) {
     root.postRun();
     return root.reporter;
   }
-  let testFiles = files ?? createTestFileList();
+  let testFiles = files ?? createTestFileList(options.globPatterns);
```

Comment:
`globPatterns` is missing input validation. I think we should also throw an exception if `globPatterns` and `files` are used together based on the current implementation.